plugins/node/opentelemetry-instrumentation-hapi/src/instrumentation.ts

```diff
@@ -337,15 +337,21 @@ export class HapiInstrumentation extends InstrumentationBase {
         const span = instrumentation.tracer.startSpan(metadata.name, {
           attributes: metadata.attributes,
         });
-        let res;
-        await api.context.with(
-          api.trace.setSpan(api.context.active(), span),
-          async () => {
-            res = await method(...params);
-          }
-        );
-        span.end();
-        return res;
+        try {
+          return await api.context.with(
+            api.trace.setSpan(api.context.active(), span),
+            () => method(...params)
+          );
```

Comment:
I think @Rauno56 was suggesting this, which avoids creating an additional closure and the overhead associated.

```suggestion
          return await api.context.with(
            api.trace.setSpan(api.context.active(), span),
            method,
            undefined,
            ...params
          );
```

Reply:
https://open-telemetry.github.io/opentelemetry-js-api/classes/contextapi.html#with

Reply:
made the change.
For some reason typescript didn't infer the generic types correctly so I had to set them manually.